clients/client-java/src/main/java/org/apache/gravitino/client/ErrorHandlers.java

```diff
@@ -858,6 +868,43 @@ public void accept(ErrorResponse errorResponse) {
     }
   }
 
+  /** Error handler specific to Credential operations. */
+  @SuppressWarnings("FormatStringAnnotation")
+  private static class CredentialErrorHandler extends RestErrorHandler {
+
+    private static final CredentialErrorHandler INSTANCE = new CredentialErrorHandler();
+
+    @Override
+    public void accept(ErrorResponse errorResponse) {
+      String errorMessage = formatErrorMessage(errorResponse);
+
+      switch (errorResponse.getCode()) {
+        case ErrorConstants.ILLEGAL_ARGUMENTS_CODE:
+          throw new IllegalArgumentException(errorMessage);
+
+        case ErrorConstants.NOT_FOUND_CODE:
+          if (errorResponse.getType().equals(NoSuchMetalakeException.class.getSimpleName())) {
+            throw new NoSuchMetalakeException(errorMessage);
+          } else if (errorResponse
```

Comment:
Em ... I don't think we want to make this exception an explicit one.
The reason is that we may intentionally make this less clear, for security's sake.

Reply:
This is out of scope of this PR as it's outside the CLI code or any changes made by this PR.  Best to raise a new issue for it.